src/digitsDFA.cpp

```diff
@@ -0,0 +1,33 @@
+#include "digitsDFA.h"
+#include "states.h"
+#include <utility>
+using namespace std;
+
+int error(char c) {
+	return DS_ERROR; // always go to error
+}
+
+int checkNonZero(char c) {
```

Comment:
maybe name it as isNonZeroDigit and isDigit --> not digging the check_____ naming convention